R/estimate_CUMINC.R

```diff
@@ -0,0 +1,89 @@
+#' Competing Events Cumulative Incidence
+#'
+#' @param CNSR Column name indicating the outcome and censoring statuses.
+#' Column must be a factor and the first level indicates censoring, the
+#' next level is the outcome of interest, and the remaining levels are the
+#' competing events.
+#' @param conf.int Confidence internal level. Default is 0.95.
+#' @param ... Additional argument passed to `tidycmprsk::cuminc()`
+#' @inheritParams estimate_KM
+#' @inheritParams visr
+#' @inheritParams add_CI.ggsurvfit
+#' @inheritParams add_risktable.ggsurvfit
+#' @inheritParams get_risktable
+#'
+#' @export
+#'
+#' @examples
+#' estimate_cuminc(
+#'   tidycmprsk::trial,
+#'   strata = "trt",
+#'   CNSR = "death_cr",
+#'   AVAL = "ttdeath"
+#' ) %>%
+#'   visr() %>%
+#'   add_CI() %>%
+#'   add_risktable(statlist = c("n.risk", "cumulative.event"))
+
+estimate_cuminc <- function(data
+                            ,strata = NULL
+                            ,CNSR
+                            ,AVAL
+                            ,conf.int = 0.95
+                            ,...){
+  # check for installation of tidycmprsk package
+  if (!"tidycmprsk" %in% rownames(utils::installed.packages()) ||
+      utils::packageVersion("tidycmprsk") < "0.1.0.9003") {
+    message("Install updated version of 'tidycmprsk' with `devtools::install_github('MSKCC-Epi-Bio/tidycmprsk')`")
+    return(invisible())
+  }
+  if (!"hardhat" %in% rownames(utils::installed.packages()) ||
+      utils::packageVersion("hardhat") <= "0.1.6") {
+    message("Install updated version of 'hardhat' with `devtools::install_github('tidymodels/hardhat')`")
+    return(invisible())
+  }
+  if (!"glue" %in% rownames(utils::installed.packages())) {
```

Comment:
We don't need glue so we can remove this

Reply:
Done! ✔️